services/worker-manager/src/providers/azure.js

```diff
@@ -500,6 +501,18 @@ class AzureProvider extends Provider {
       throw error();
     }
 
+    // verify the subject of the signing certificate
+    const signerCommonName = crt.subject.getField({ name: 'commonName' });
+    if (!(signerCommonName &&
+          (signerCommonName.value === 'metadata.azure.com' ||
+           signerCommonName.value.endsWith('.metadata.azure.com')))) {
+      this.monitor.log.registrationErrorWarning({
```

Comment:
The registered report here is nice but I think this might be easier later if we use `monitor.reportError`. That way all of this ends up in the sentry report. Either that or log like this _and_ attach `crt.subject` etc to the error?

Reply:
I'm not familiar enough with ``monitor`` to understand your suggestion. Can you suggest a code change?

I tried to follow the conventions I could see in this code:

* Log the error details with ``registrationErrorWarning``:
  - Use a constant string for the ``message``
  - Add details in ``error``
* Throw the generic ``error()``, initialized near the start of ``registerWorker``, to avoid leaking information to an attacker

Reply:
I found the `library/monitor` docs, I understand your comment now.

Maybe the pattern of "log a ``registerionErrorWarning``, throw a generic ``error``" should be moved to a function, which could also send the info to sentry with ``monitor.reportError``. That would change the existing code as well, but would remove a lot of repetition.

All of this code is either a trap for hackers, or a obstacle course for maintainers. Or maybe both! Depending on which way you see it, a log message and/or a Sentry report makes sense.